src/dms-preview/azext_dms/custom.py

```diff
@@ -162,42 +170,150 @@ def create_task(
 
 
 def cutover_sync_task(
+        cmd,
         client,
         resource_group_name,
         service_name,
         project_name,
         task_name,
-        database_name):
-    sync_input = MigrateSyncCompleteCommandInput(database_name=database_name)
-
+        object_name=None,
+        immediate=False):
+    # If object name is empty, treat this as cutting over the entire online migration.
+    # Otherwise, for scenarios that support it, just cut over the migration on the specified object.
     # 'input' is a built in function. Even though we can technically use it, it's not recommended.
     # https://stackoverflow.com/questions/20670732/is-input-a-keyword-in-python
-    sync_properties_params = {'input': sync_input}
-    sync_properties = MigrateSyncCompleteCommandProperties(**sync_properties_params)
 
-    client.command(group_name=resource_group_name,
-                   service_name=service_name,
-                   project_name=project_name,
-                   task_name=task_name,
-                   parameters=sync_properties)
+    src, tgt = get_project_platforms(cmd,
+                                     project_name=project_name,
+                                     service_name=service_name,
+                                     resource_group_name=resource_group_name)
+    st = get_scenario_type(src, tgt, "onlinemigration")
+
+    if st in [ScenarioType.mysql_azuremysql_online, ScenarioType.postgres_azurepostgres_online]:
+        command_input = MigrateSyncCompleteCommandInput(database_name=object_name)
+        command_properties_model = MigrateSyncCompleteCommandProperties
+    elif st == ScenarioType.mongo_mongo_online:
+        command_input = MongoDbFinishCommandInput(object_name=object_name, immediate=immediate)
+        command_properties_model = MongoDbFinishCommand
+
+    run_command(client,
+                command_input,
+                command_properties_model,
+                resource_group_name,
+                service_name,
+                project_name,
+                task_name)
+
+
+def restart_task(
+        cmd,
+        client,
+        resource_group_name,
+        service_name,
+        project_name,
+        task_name,
+        task_type,
+        object_name=None):
+    # For scenarios that support it, restart the entire migration if object name is empty,
+    # otherwise restart the specified object.
+    src, tgt = get_project_platforms(cmd,
+                                     project_name=project_name,
+                                     service_name=service_name,
+                                     resource_group_name=resource_group_name)
+    st = get_scenario_type(src, tgt, task_type)
```

Comment:
Do we need to prompt the user the task type? Can we just fail if the user uses a scenario that isn't supported?

Reply:
MongoDB migrations will allow restarting both online and offline migrations. This might not be the case with other scenarios. Also, right now, the extension doesn't support online MongoDB migrations. So I think this needs to be left there. Unless we can come up with a work-around to using task type in general.

Reply:
Thinking about this more, you might be right. The task is already created and the different combinations are filtered out there already. I think I will remove task type here as well. If it ends up that we need it later it should be easy to add back.